lib/index.js

```diff
@@ -156,43 +178,42 @@ internals.vision = class Vision {
     }
 
 
-    getState(realm) {
-
-        const state = realm.plugins.vision = realm.plugins.vision || {};
-        return state;
-    }
-
-
-    nearestManager(realm) {
+    getRootRealm(realm) {
 
-        const pluginState = this.getState(realm);
-        if (pluginState.manager) {
-            return pluginState.manager;
+        while (realm.parent) {
+            realm = realm.parent;
         }
 
-        let parent = realm.parent;
-        while (parent) {
-
-            const parentState = this.getState(parent);
-            if (parentState.manager) {
-                return parentState.manager;
-            }
+        return realm;
+    }
+};
 
-            parent = parent.parent;
-        }
 
-        return null;
-    }
+exports.getManager = (ref) => {
```

Comment:
Curious!  I notice other hapi libraries utilizing request/sever/toolkit decorations for this type of thing.